src/command_modules/azure-cli-storage/azure/cli/command_modules/storage/_factory.py

```diff
@@ -83,9 +90,10 @@ def queue_data_service_factory(kwargs):
         connection_string=kwargs.pop('connection_string', None),
         sas_token=kwargs.pop('sas_token', None))
 
+
 def cloud_storage_account_service_factory(kwargs):
     account_name = kwargs.pop('account_name', None)
     account_key = kwargs.pop('account_key', None)
     sas_token = kwargs.pop('sas_token', None)
-    connection_string = kwargs.pop('connection_string', None) # pylint: disable=unused-variable
+    kwargs.pop('connection_string', None)  # pylint: disable=unused-variable
```

Comment:
I believe only the 'storage account sas create' method uses this, but here we should actually parse the connection string into account name and key if it was provided and they were not. However, that may already happen somewhere else and thus simply popping the connection string is fine. 

If you can run 'az storage account sas create...' by specifying `--connection-string {foo}` then this change is fine. Otherwise please do the parsing.